internal/worker/instancepoller/manifold.go

```diff
@@ -14,16 +14,21 @@ import (
 
 	"github.com/juju/juju/api/base"
 	"github.com/juju/juju/api/controller/instancepoller"
+	coredependency "github.com/juju/juju/core/dependency"
 	"github.com/juju/juju/core/instance"
 	"github.com/juju/juju/core/logger"
 	"github.com/juju/juju/core/network"
 	"github.com/juju/juju/core/watcher"
 	"github.com/juju/juju/environs"
 	"github.com/juju/juju/environs/envcontext"
 	"github.com/juju/juju/environs/instances"
+	"github.com/juju/juju/internal/servicefactory"
 	"github.com/juju/juju/internal/worker/common"
 )
 
+// GetMachineFunc is a helper function that gets a service from the manifold.
+type GetMachineServiceFunc func(getter dependency.Getter, name string) (MachineService, error)
+
 // facadeShim wraps an instancepoller API instance and allows us to provide
 // methods that return interfaces which we can easily mock in our tests.
 type facadeShim struct {
```

Comment:
The facadeShim still implements WatchModelMachines(), as does the hand written mockFacadeAPI.